### ISSUE TITLE
ICMP packets larger than interface MTU are not correctly processed

### ISSUE THREAD

**Issue by mato** [mirage/mirage-tcpip#237]
To reproduce, pick any unikernel with a network stack. Send it an ICMP echo packet larger than the interface MTU, e.g. `ping -s 2000 -c 1 10.0.0.2`.
Unikernel output:

```
2016-08-29 15:17:50 +02:00: INF [icmpv4] ICMP: error parsing message from 10.0.0.1: unrecognized ICMPv4 type
```

`ping` output:

```
PING 10.0.0.2 (10.0.0.2) 2000(2028) bytes of data.
1480 bytes from 10.0.0.2: icmp_seq=1 ttl=38 (truncated)

--- 10.0.0.2 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.231/0.231/0.231/0.000 ms
```

Reproducible on (at least) `unix`, `virtio` and `ukvm` targets so looks like an ICMP issue.

**Comment by polvorin** [comment]
Hi, the problem is because mirage-tcpip  doesn't try to handle IP fragmentation.  
Actually I was planning to add that and prepare a pull request,  depending on how much free time I've  might be ready over the next days.

**Comment by yomimono** [comment]
IP fragmentation has been a `TODO` in tcpip for ages; it would be nice to have a patch.  Testing ICMP directly shows that it should be able to handle the large payload if the underlying layer can pass it up properly.

**Comment by hannesm** [comment]
ip reassembly is handled not in #375, no fragmentation yet though ;) (it's on the concrete agenda, but requires fixes at the lowest layer -- see my proposed mirage-net change to expose its mtu https://github.com/mirage/mirage-net/pull/13)

**Comment by hannesm** [comment]
fixed in master since #394
